From 1bcd127f1116fb3a5f47cb7fc675954d97860235 Mon Sep 17 00:00:00 2001
From: Kevin Cernekee <cernekee@gmail.com>
Date: Mon, 29 Jan 2018 00:32:22 -0800
Subject: [PATCH] HACK: Fix Android build

Bionic stdio internals are opaque, and not static from one release to
the next.  Add a couple of hacks to allow building liboath with the
latest NDK toolchains.
---

diff --git a/liboath/gl/stdio-impl.h b/liboath/gl/stdio-impl.h
index 4c02c9f614ba..47178be99e94 100644
--- a/liboath/gl/stdio-impl.h
+++ b/liboath/gl/stdio-impl.h
@@ -58,6 +58,20 @@
 #  define fp_ fp
 # endif
 
+#if defined __ANDROID__
+/* This is probably wrong, but it's enough to get things to compile again.
+ * Android does not expose stdio internals, and the structs vary from one
+ * release to the next. */
+struct __sbuf {
+  unsigned char* _base;
+#if defined(__LP64__)
+  size_t _size;
+#else
+  int _size;
+#endif
+};
+#endif
+
 # if (defined __NetBSD__ && __NetBSD_Version__ >= 105270000) || defined __OpenBSD__ || defined __ANDROID__ /* NetBSD >= 1.5ZA, OpenBSD, Android */
   /* See <http://cvsweb.netbsd.org/bsdweb.cgi/src/lib/libc/stdio/fileext.h?rev=HEAD&content-type=text/x-cvsweb-markup>
      and <http://www.openbsd.org/cgi-bin/cvsweb/src/lib/libc/stdio/fileext.h?rev=HEAD&content-type=text/x-cvsweb-markup> */
-- 
2.7.4

